py-polars/polars/internals/frame.py

```diff
@@ -1638,6 +1652,59 @@ def _pos_idx(self, idx: int, dim: int) -> int:
         else:
             return self.shape[dim] + idx
 
+    def _pos_idxs(self, idxs: np.ndarray | pli.Series, dim: int) -> pli.Series:
+        if isinstance(idxs, pli.Series):
+            if idxs.dtype == UInt32:
+                return idxs
+            if idxs.dtype in {
+                UInt8,
+                UInt16,
+                UInt64,
+                Int8,
+                Int16,
+                Int32,
+                Int64,
+            }:
+                if idxs.dtype in {Int64, UInt64}:
+                    if idxs.max() >= 2**32:  # type: ignore[operator]
```

Comment:
This is not true in polars-big-idx.

Reply:
Maybe we can extract the python package name somehow?

Reply:
`pl.__package__` seems to be `polars` in both cases.

Maybe you can set the index size in rust and request it with `pl.Config.get_idx_size()` or with `pl.Idx`.

Reply:
I just remembered that there is a hacky way to detect it as some dtype supertype promotions (fixed in https://github.com/pola-rs/polars/pull/3329 ) are different.

```python
# in normal polars
In [15]: (pl.Series("", [1], dtype=pl.UInt64) + pl.Series("", [1], dtype=pl.Int64)).dtype == pl.Float64
Out[15]: True

# In polars_u64_idx
In [14]: (pl.Series("", [1], dtype=pl.UInt64) + pl.Series("", [1], dtype=pl.Int64)).dtype == pl.Int64
Out[14]: True
```

Reply:
We can export it from the binary. Let me make a function for that and then we can rebase this upon that work?

Reply:
@ghuls see: #4167

Reply:
Just pushed the changes to support u64 indexes.